Sources/Fuzzilli/FuzzIL/JsOperations.swift

```diff
@@ -1047,21 +1047,37 @@ final class TestIn: JsOperation {
 
 }
 
+enum ParameterPattern {
+    case identifier
+    case object(properties: [ObjectPatternProperty])
```

Comment:
Could this also be a `[String: ParameterPattern]` dictionary or would that make things more complicated? (asking because that "feels" a bit more natural without having thought about it deeply)

Reply:
Unfortunately switching to a dictionary based approach causes problem because it does not contain information about the order of the object property keys. The dictionary tells us which value belongs to which key but not in which order the keys appear in the object pattern. This is why I opted for an array.

Reply:
Technically yes. The order of the parameters doesn't matter. However the mapping from parameter -> variable is lost this way (IIUC). I.e. the order of variables is fixed but the order of object properties is permutated. 

We can do it anyway by using an array of tuples instead of a dictionary. This way we can retain the orderedness.

Reply:
Right yeah that's a good point. It also needs to be consistent across serialization and deserialization (see https://github.com/googleprojectzero/fuzzilli/pull/449#issuecomment-2468034053) so I guess a list of tuples is the way to go here 👍